src/mode/modeVisual.ts

```diff
@@ -0,0 +1,140 @@
+"use strict";
+
+import * as _      from 'lodash';
+
+import { ModeName, Mode } from './mode';
+import { Motion} from './../motion/motion';
+import { Position } from './../motion/position';
+import { Operator } from './../operator/operator';
+import { DeleteOperator } from './../operator/delete';
+import { ModeHandler } from './modeHandler.ts';
+import { ChangeOperator } from './../operator/change';
+
+export class VisualMode extends Mode {
+    /**
+     * The part of the selection that stays in the same place when motions are applied.
+     */
+    private _selectionStart: Position;
+
+    /**
+     * The part of the selection that moves.
+     */
+    private _selectionStop : Position;
+    private _modeHandler   : ModeHandler;
+
+    private _keysToOperators: { [key: string]: Operator };
+
+    constructor(motion: Motion, modeHandler: ModeHandler) {
+        super(ModeName.Visual, motion);
+
+        this._modeHandler = modeHandler;
+        this._keysToOperators = {
+            // TODO: use DeleteOperator.key()
+
+            // TODO: Don't pass in mode handler to DeleteOperators,
+            // simply allow the operators to say what mode they transition into.
+            'd': new DeleteOperator(modeHandler),
+            'x': new DeleteOperator(modeHandler),
+            'c': new ChangeOperator(modeHandler)
+        };
+    }
+
+    shouldBeActivated(key: string, currentMode: ModeName): boolean {
+        return key === "v";
+    }
+
+    async handleActivation(key: string): Promise<void> {
+        this._selectionStart = this.motion.position;
```

Comment:
This is great first cut. Would it be worthwhile to move selection to it's own class? That is what I had envisioned for the `motion` class.